pxr/usdImaging/usdImaging/delegate.cpp

```diff
@@ -1348,10 +1348,35 @@ UsdImagingDelegate::_ResyncUsdPrim(SdfPath const& usdPath,
     // Otherwise, this is a totally new branch of the scene, so populate
     // from the resync target path. Note: we need to verify that usdPath exists,
     // since we can get resync notices for prims that were deleted.
-    if (_stage->GetPrimAtPath(usdPath)) {
+    UsdPrim usdPrim = _stage->GetPrimAtPath(usdPath);
+    if (usdPrim) {
         TF_DEBUG(USDIMAGING_CHANGES).Msg("  - affected new prim: <%s>\n",
             usdPath.GetText());
         proxy->Repopulate(usdPath);
+
+        if (usdPrim.HasAPI<UsdLuxLightAPI>()) {
+            // If a new light was added, ensure collections are dirtied
+            TRACE_SCOPE("Resync light/shadow link targets");
+            auto _MarkLinkPathsDirty = [&](const SdfPathSet& linkedPaths) {
+                for (SdfPath affectedCachePath : linkedPaths) {
+                    TF_DEBUG(USDIMAGING_CHANGES).Msg("[Refresh Object]: "
+                        "Light <%s> modified; invalidate linked prim %s\n", 
+                        usdPath.GetText(), affectedCachePath.GetText());
+                    _HdPrimInfo* primInfo = _GetHdPrimInfo(affectedCachePath);
+                    if (primInfo && primInfo->usdPrim.IsValid() &&
+                        TF_VERIFY(primInfo->adapter, "%s", affectedCachePath.GetText())) {
+                        UsdImagingPrimAdapterSharedPtr& adapter = primInfo->adapter;
+                        adapter->MarkDirty(primInfo->usdPrim, affectedCachePath,
+                            HdChangeTracker::DirtyCategories, proxy);
+                    }
+                }
+            };
+            UsdLuxLightAPI light(usdPrim);
+            UsdCollectionAPI lightLink{ light.GetLightLinkCollectionAPI() };
+            _MarkLinkPathsDirty(lightLink.ComputeIncludedPaths(lightLink.ComputeMembershipQuery(), _stage));
```

Comment:
I think the "Compute" here won't go through the collection cache, right?  Is there a way to do that?

Reply:
This code is now deleted